### PR TITLE
[PIM-6574+PIM-6555] Products index

### PR DESCRIPTION
**Pull request by pierallard**
This PR moves elements to a left column in the products index:
- locale switcher
- scope switcher
- category switcher
- view switcher

In fact, this is a mix of all these cards.
PIM-6574
PIM-6555
PIM-6505
PIM-6524
PIM-6575
PIM-6553
PIM-6514
PIM-6515
PIM-6587

- [x] Catalog switcher: display on column
- [x] Review JS tree font size and alignments
- [x] Scope switcher: display on initialization (it works only on change)*
- [x] Scope switcher: it does not remains on reload!
- [ ] Bug: come on products / come on published products / get back / => 0 products on the category tree!
- [x] Fix 3rd column on scroll
- [x] Check the design work with a loooooong project name

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added Behats                      | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

### CODE REVIEW

File: features/Context/WebUser.php [hunk 1]
```diff
@@ -396,6 +396,33 @@ function () use ($pageName, $data, $copy) {
...
+
+            $categorySwitcher = $this->getCurrentPage()->find('css', '.category-switcher');
+            if (null !== $categorySwitcher) {
+                $categorySwitcher->click();
+            }
+
+            return false;
+        }, 'Cannot find the category switcher');
+    }
+
```
Comment (pierallard) [inline features/Context/WebUser.php]: DOBBLE LINEEEEE

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/grid/column-category.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                this.$el.html(this.template({
+                    categoryLabel: __('pim_enrich.entity.product.category'),
+                    isHighlited: this.isHighlited
+                }));
+
+                this.renderExtensions();
+            },
+
+            /**
+             * Toggle the thrid column
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/grid/column-category.js]: thrid

File: src/Pim/Bundle/UIBundle/Resources/public/less/lib/select2.less [hunk 3]
```diff
@@ -216,139 +216,127 @@
...
       }
     }
   }
 
   &.select2-drop {
-    width: 400px !important; // The select2 lib sets the width with JS methods.
+    box-shadow: 1px 2px 8px rgba(0,0,0,0.15);
+    margin-top: -45px;
+    margin-left: -10px;
+    min-width: 300px;
```
Comment (pierallard) [inline src/Pim/Bundle/UIBundle/Resources/public/less/lib/select2.less]: should be 340

File: src/Pim/Bundle/UIBundle/Resources/public/less/lib/select2.less [hunk 4]
```diff
@@ -381,7 +369,7 @@
     border: 1px solid @AknBorderColor !important;
     padding: 5px;
     margin-top: 5px;
-    min-width: 300px;
+    min-width: 340px;
```
Comment (pierallard) [inline src/Pim/Bundle/UIBundle/Resources/public/less/lib/select2.less]: revert this

File: src/Pim/Bundle/UIBundle/Resources/public/less/lib/select2.less (hunk 3 above)
Comment (jjanvier) [reply src/Pim/Bundle/UIBundle/Resources/public/less/lib/select2.less]: :ah:

File: src/Pim/Bundle/EnrichBundle/Resources/config/form_extensions/product/index.yml [hunk 5]
```diff
@@ -5,32 +5,50 @@ extensions:
...
             template: pim/template/product/index
             templateParams:
-                title: Products
+                titleKey: pim_enrich.entity.product.index_title
                 gridName: product-grid
 
+    pim-product-index-left-column:
+        module: pim/form/common/column
+        parent: pim-product-index
+        position: 5
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Resources/config/form_extensions/product/index.yml]: drop it

File: src/Pim/Bundle/EnrichBundle/Resources/config/requirejs.yml [hunk 6]
```diff
@@ -184,6 +184,7 @@ config:
                         urls:
                             list: pim_enrich_category_rest_list
                             get: pim_enrich_category_rest_get
+                            search: pim_enrich_category_rest_search
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Resources/config/requirejs.yml]: remove it

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/grid/category-switcher.js [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+            },
+            isHighlited: false,
+            categoryLabel: null,
+            treeLabel: null,
+
+            /**
+             * {@inheritdoc}
+             */
+            configure: function () {
+                this.listenTo(this.getRoot(), 'pim_enrich:form:category_updated', this.updateValue);
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/grid/category-switcher.js]: move the value to have directly the labels

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/grid/column-inner.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+        return BaseForm.extend({
+            template: _.template(template),
+
+            /**
+             * {@inheritdoc}
+             */
+            render() {
+                this.$el.html(this.template);
+
+                this.renderExtensions();
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/grid/column-inner.js]: simple view

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/grid/category-switcher.js (hunk 7 above)
Comment (pierallard) [reply src/Pim/Bundle/EnrichBundle/Resources/public/js/product/grid/category-switcher.js]: done

**Pull request merged**